src/whitelist.ts

```diff
@@ -0,0 +1,98 @@
+import { NonceManager } from '@ethersproject/experimental';
+import { ContractTransaction, ethers } from 'ethers';
+import groupBy from 'lodash/groupBy';
+import { Choice, PromptObject } from 'prompts';
+import { runAndHandleErrors } from './utils/cli';
+import { getDapiServerContract } from './utils/evm';
+import { loadCredentials } from './utils/filesystem';
+import { promptQuestions } from './utils/prompts';
+import { readOperationsRepository } from './utils/read-operations';
+
+const questions = (choices: Choice[]): PromptObject[] => {
+  return [
+    {
+      type: 'autocompleteMultiselect',
+      name: 'selectedSubscriptions',
+      message: 'Please select subscriptions for which to whitelist an address to read from DapiServer contract?',
+      choices: choices,
+    },
+    {
+      type: 'text',
+      name: 'mnemonic',
```

Comment:
we should probably use the credentials file instead of using a single mnemonic. Might be the case that the mnemonic is different for each chain. You can build nounceManager for each chain similar to [this](https://github.com/api3dao/operations/blob/1e57427f8cad67521a3aa9d97c89e7f3c5c407b4/src/register-subscriptions.ts#L33) .

Reply:
Even though we will probably only use a single mnemonic across all chains available chains, I see your point 👍🏻